ui/src/copyable_text.tsx

```diff
@@ -54,19 +64,25 @@ export interface CopyableText {
   height?: S
 }
 
-export const XCopyableText = ({ model }: { model: CopyableText }) => {
+type ClipboardCopyButton = {
+  /** Text to be copied to clipboard. */
+  value: S,
+  /** The element to which the copy button is attached. */
+  anchorElement: HTMLElement | undefined,
+  /** Show copy button only on hover over anchor element. */
+  showOnHoverOnly?: B
+}
+
+export const ClipboardCopyButton = ({ value, anchorElement, showOnHoverOnly = false }: ClipboardCopyButton) => {
   const
-    { name, multiline, label, value, height } = model,
-    heightStyle = multiline && height === '1' ? fullHeightStyle : undefined,
-    ref = React.useRef<Fluent.ITextField>(null),
     timeoutRef = React.useRef<U>(),
     [copied, setCopied] = React.useState(false),
-    onClick = async () => {
-      const el = ref.current
-      if (!el) return
+    [visible, setVisible] = React.useState(!showOnHoverOnly),
+    onClick = React.useCallback(async () => {
+      if (!anchorElement) return
```

Comment:
Can the copy to clipboard be done without `anchorElement`?

Reply:
In order for _copy exec command_ to work we need to call `select()` on input/textarea element.
```tsx
if (document.queryCommandSupported('copy')) {
          (anchorElement as any)?.select()
          document.execCommand('copy')
          window.getSelection()?.removeAllRanges()
        }
```
Therefore we need to have its reference inside ClipboardCopyButton component. 

If we don't want to use `anchorElement` prop, the option would be to obtain its reference internally inside the ClipboardCopyButton component. But how? The general solution won't work (e.g. getting the parent of ClipboardCopyButton) because
- for `copyable_text` the button is attached to label, so textarea/input is the sibling of the button container 
- for markdown code block the copy button is a child of the code block

In this case what comes to my mind is to make the ClipboardCopyButton component a wrapper and the input/textarea/code will always be a child to ClipboardCopyButton component.
```tsx
<ClipboardCopyButtonWrapper>
   [markdown_code/copyable_text_input]
</ClipboardCopyButtonWrapper>
```

The problem with this approach is that FluentUI does not provide the option to override the input element for multiline input of `Fluent.TextField` so we couldn't wrap it inside a ClipboardCopyButtonWrapper.

Reply:
> In order for copy exec command to work we need to call select() on input/textarea element.

Correct, but why jump through hoops when the element can simply be created on the fly, its value set, copied and then removed? Since this is a common problem, a quick prompt to copilot returns:

```js
function copyToClipboard(str) {
  const el = document.createElement('textarea');
  el.value = str;
  document.body.appendChild(el);
  el.select();
  document.execCommand('copy');
  document.body.removeChild(el);
}
```

Reply:
What a nice trick! Thanks @mturoci .